forge-game/src/main/java/forge/game/GameAction.java

```diff
@@ -747,24 +748,23 @@ public final Card moveTo(final Zone zoneTo, Card c, SpellAbility cause, Map<Abil
         // use FThreads.invokeInNewThread to run code in a pooled thread
         return moveTo(zoneTo, c, null, cause, params);
     }
-    public final Card moveTo(final Zone zoneTo, Card c, Integer position, SpellAbility cause) {
-        return moveTo(zoneTo, c, position, cause, AbilityKey.newMap());
-    }
 
     public final Card moveTo(final ZoneType name, final Card c, SpellAbility cause, Map<AbilityKey, Object> params) {
         return moveTo(name, c, 0, cause, params);
     }
-    public final Card moveTo(final ZoneType name, final Card c, final int libPosition, SpellAbility cause) {
-        return moveTo(name, c, libPosition, cause, AbilityKey.newMap());
-    }
+
     public final Card moveTo(final ZoneType name, final Card c, final int libPosition, SpellAbility cause, Map<AbilityKey, Object> params) {
         // Call specific functions to set PlayerZone, then move onto moveTo
         switch(name) {
             case Hand:          return moveToHand(c, cause, params);
             case Library:       return moveToLibrary(c, libPosition, cause, params);
             case Battlefield:   return moveToPlay(c, c.getController(), cause, params);
             case Graveyard:     return moveToGraveyard(c, cause, params);
-            case Exile:         return exile(c, cause, params);
+            case Exile:
+                if (!c.canExiledBy(cause, true)) {
```

Comment:
because inside the exile function i can't check if gets exiled by cost or by effect

Reply:
Otherwise i need to stop calling the exile function from moveZone

Reply:
would anything even get stopped from that check there?
The way it's written each effect needs to consider the Static by itself anyway so it can adjust its logic
Same for the costs

Reply:
Costs that exile doesn't call the "moveZone" function

and effects that exile are checked before too.

its mostly just a fail-safe, to prevent it from causing trouble